Move asset evaluation logic out of SDK

uranusjr opened the pull request:
Asset evaluation is only done in the scheduler, and requires the database in various cases. It is better to split it out into a dedicated class in Airflow core.

Close #47378
Fix #47483

uranusjr committed:
Move asset evaluation logic out of SDK

Asset evaluation is only done in the scheduler, and requires the
database in various cases. It is better to split it out into a dedicated
class in Airflow core.

ashb reviewed (approved):
That was quick, thanks!

uranusjr committed:
Remove dependency resolution in DAG parser

Dependency resolution requires database calls for non-direct asset
references, but we don't want that to happen in the SDK. This removes
the eager resolution code so all asset aliases and refs are no longer
resolved, but one only keeps a marker for itself.

An additional PR will be submitted later to do the resolution. This is a
part of the asset UI project that involves changes in the API anyway.
